spec.html

```diff
@@ -15465,19 +15465,118 @@ <h1>Runtime Semantics: ApplyNumericBinaryOperator ( _lval_, _opText_, _rval_ )</
         1. Let _rnum_ be ? ToNumeric(_rval_).
         1. If Type(_lnum_) is different from Type(_rnum_), throw a *TypeError* exception.
         1. Let _T_ be Type(_lnum_).
-        1. If _opText_ is `**`, return ? _T_::exponentiate(_lnum_, _rnum_).
-        1. If _opText_ is `*`, return _T_::multiply(_lnum_, _rnum_).
-        1. If _opText_ is `/`, return _T_::divide(_lnum_, _rnum_).
-        1. If _opText_ is `%`, return _T_::remainder(_lnum_, _rnum_).
-        1. If _opText_ is `+`, return _T_::add(_lnum_, _rnum_).
-        1. If _opText_ is `-`, return _T_::subtract(_lnum_, _rnum_).
-        1. If _opText_ is `&lt;&lt;`, return _T_::leftShift(_lnum_, _rnum_).
-        1. If _opText_ is `&gt;&gt;`, return _T_::signedRightShift(_lnum_, _rnum_).
-        1. If _opText_ is `&gt;&gt;&gt;`, return _T_::unsignedRightShift(_lnum_, _rnum_).
-        1. If _opText_ is `&amp;`, return _T_::bitwiseAND(_lnum_, _rnum_).
-        1. If _opText_ is `|`, return _T_::bitwiseOR(_lnum_, _rnum_).
-        1. If _opText_ is `^`, return _T_::bitwiseXOR(_lnum_, _rnum_).
-        1. Assert: Execution cannot reach this step.
+        1. Let _result_ be the value determined according to the following table:
+          <figure>
+            <table class="lightweight-table">
+              <tbody>
+                <tr>
+                  <th>
+                    _opText_
+                  </th>
+                  <th>
+                    _result_
+                  </th>
+                </tr>
+                <tr>
+                  <td>
+                    `**`
+                  </td>
+                  <td>
+                    _T_::exponentiate(_lnum_, _rnum_)
```

Comment:
I would've just had this column contain the AO (`_T_::exponentiate`) and then in a following step call `_result_(_lnum_, _rnum_)`. Also, a better name would then be `_compute_` or something. Like this, it's just not super clear when/if the AOs in this table are called.

Reply:
> I would've just had this column contain the AO (`_T_::exponentiate`) and then in a following step call `_result_(_lnum_, _rnum_)`. Also, a better name would then be `_compute_` or something.

Okay, I've called it `_operation_`.

> Like this, it's just not super clear when/if the AOs in this table are called.

Which suggests to me that an inline table isn't *really* EcmaSpeak's exhaustive switch construct, because a switch construct wouldn't have that lack of clarity. E.g., in a programming language, if you saw
```
case "**": result = exponentiate(lnum, rnum);
```
or
```
when "**" -> exponentiate(lnum, rnum),
```
there wouldn't be confusion about when/if 'exponentiate' is called.

Instead, it sounds like inline tables are for (more-or-less) "static" content -- looking up predetermined values without involving any execution.

Reply:
> Instead, it sounds like inline tables are for (more-or-less) "static" content -- looking up predetermined values without involving any execution.

Fair, but you can always represent behaviour as data, then act on it later. Seems sufficient for our limited needs.

Reply:
> you can always represent behaviour as data

So, e.g., a table-cell could define an abstract closure that wraps the desired behaviour?